x-pack/plugins/lists/server/routes/summary_exception_list_route.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import {
+  SummaryExceptionListSchemaDecoded,
+  summaryExceptionListSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { EXCEPTION_LIST_URL } from '@kbn/securitysolution-list-constants';
+
+import type { ListsPluginRouter } from '../types';
+
+import {
+  buildRouteValidation,
+  buildSiemResponse,
+  getErrorMessageExceptionList,
+  getExceptionListClient,
+} from './utils';
+
+export const summaryExceptionListRoute = (router: ListsPluginRouter): void => {
+  router.get(
+    {
+      options: {
+        tags: ['access:lists-summary'],
+      },
+      path: `${EXCEPTION_LIST_URL}/_summary`,
+      validate: {
+        query: buildRouteValidation<
+          typeof summaryExceptionListSchema,
+          SummaryExceptionListSchemaDecoded
+        >(summaryExceptionListSchema),
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+      try {
+        const { id, list_id: listId, namespace_type: namespaceType } = request.query;
+        const exceptionLists = getExceptionListClient(context);
+        if (id != null || listId != null) {
+          const exceptionListSummary = await exceptionLists.getExceptionListSummary({
+            id,
+            listId,
+            namespaceType,
+          });
+          if (exceptionListSummary == null) {
+            return siemResponse.error({
+              body: getErrorMessageExceptionList({ id, listId }),
+              statusCode: 404,
+            });
+          } else {
+            return response.ok({ body: exceptionListSummary ?? {} });
```

Comment:
You can optionally validate responses too if that makes you feel like you will have a better chance of keeping your code working long term. We have some examples like this line here:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/lists/server/routes/delete_list_item_route.ts#L47


Basically, if you define both a schema for request and one for response, you can validate both the request and the response. You can do a "strict" check or a looser non-strict check for the response. Usually I try to validate both request and response using strict checks like that above example and then I write an e2e test that exercises my route.

It makes it to where if someone changes the contract by accident by say adding an extra key/value on the response, it would fail in the validation logic during the e2e test and prevent them from checking in until they fixed the response validation schema.

Reply:
Got it! I will add those validations and also add a new e2e test! Thanks